boards/arduino-uno/examples/uno-watchdog.rs

```diff
@@ -0,0 +1,17 @@
+#![no_std]
+#![no_main]
+
+use arduino_uno::prelude::*;
+use panic_halt as _;
+
+#[arduino_uno::entry]
+fn main() -> ! {
+    let dp = arduino_uno::Peripherals::take().unwrap();
+    let mut watchdog = arduino_uno::wdt::Wdt::new(&dp.CPU.mcusr, dp.WDT);
+    watchdog.disable();
```

Comment:
Is `disable()` before `start()` necessary or is this just for demonstration?

Reply:
It sounds like it's to catch a (rare?) case where the watchdog is accidentally enabled. They suggest always disabling the watchdog on init, even if the watchdog isn't being used. From the data sheet:

> Note: If the watchdog is accidentally enabled, for example by a runaway pointer or brown-out condition, the
> device will be reset and the watchdog timer will stay enabled. If the code is not set up to handle the watchdog,
> this might lead to an eternal loop of time-out resets. To avoid this situation, the application software should
> always clear the watchdog system reset flag (WDRF) and the WDE control bit in the initialization routine, even
> if the watchdog is not in use.

Reply:
Hm, I'm reading this a bit differently:  When you're not using the watchdog, you're supposed to disable it in your init code anyway, just in case it got accidentally enabled.  Otherwise it will stay enabled and cause an eternal loop of resets.  When you're actively using the watchdog (as you do in the example), it's not needed because you're configuring and feeding it properly.

So to me this says e.g. a bootloader or runtime should clear WDRF and WDE on startup.  But to code using the watchdog as intended, it's not relevant.

Reply:
I removed the disable() call from the examples.